scripts/partition_manager.py

```diff
@@ -46,6 +52,8 @@ def empty_one_of(one_of_list):
     for k, v in reqs.items():
         if isinstance(v, dict):
             if 'one_of' in v.keys():
+                if k in ('after', 'before'):
```

Comment:
I refactored this change to support an `invalid` keyword in the function, which should make this be even more future-proof than desired here. The error message was also updated to reflect this.

The reasoning was that because of the recursion, we would need some way of continually tracking whether or not `placement` was a parent key somewhere along the way. If we had `placement: { before: [ ] }`, then the parent key is easy to check. It is not so easy if a feature is added down the line that makes us have to check `placement: {something1: {something2: [] }}`, in which case a simple parent check would be `something1` instead of `placement`.